Library/ViewModels/PostCampaignCheckoutViewModel.swift

```diff
@@ -231,6 +238,53 @@ public class PostCampaignCheckoutViewModel: PostCampaignCheckoutViewModelType,
     self.showErrorBannerWithMessage = Signal
       .combineLatest(validateCheckoutExistingCard.errors(), validateCheckoutNewCard.errors())
       .map { _ in Strings.Something_went_wrong_please_try_again() }
+
+    let paymentAuthorizationData: Signal<PaymentAuthorizationData, Never> = self.configureWithDataProperty
+      .signal
+      .skipNil()
+      .map { (data: PostCampaignCheckoutData) -> PaymentAuthorizationData in
+        let reward = data.rewards.first! // What happens if you select no reward?
+        return PaymentAuthorizationData(
+          project: data.project,
+          reward: reward,
+          // TODO: PKPaymentRequests+Helper switches on reward.isNoReward for displaying the total
+          allRewardsTotal: reward.isNoReward ? 0 : data.total,
+          additionalPledgeAmount: reward.isNoReward ? data.total : 0,
```

Comment:
I think this is an example of how keeping two separate flows is adding tech debt. In the old checkout flow, the "no rewards" case reward never gets updated, so we need to check the "bonus" amount to find out what amount is currently associated with it. But in the new flow I updated the code that configures the data for the checkout screen to update the "no rewards" reward with its final amount (and set the bonus to 0 so it doesn't get counted twice) in order to make it display nicely in the table without needing the table to know about the "no rewards" case. So this might be something that we just leave kind of ugly until we manage to unify our checkout flows.

Reply:
Agreed, with regards to the tech debt. And thanks for the explanation - I was kind of confused how I ended up in this position, so it's interesting to hear what the root cause is.

It also doesn't help that the code to create the ApplePay configuration takes these (somewhat opaque) parameters; it's unclear when you examine the method signature what it's actually going to do with the various values.